dashboard/routes/wallet/create.go

```diff
@@ -33,17 +36,73 @@ func WalletCreateRouter(router *gin.RouterGroup) {
 			log.WebLog.Error("wallet file exists.")
 			context.AbortWithError(http.StatusInternalServerError, errors.New("wallet file exists."))
 			return
-		} else {
-			_, err := vault.NewWallet(config.Parameters.WalletFile, []byte(data.Password), true)
+		}
+
+		if config.Parameters.WebGuiCreateWallet && !config.Parameters.AllowEmptyBeneficiaryAddress {
+			if data.BeneficiaryAddr == "" {
```

Comment:
I think we should move `!config.Parameters.AllowEmptyBeneficiaryAddress` from the previous line to this line, so that even if `config.Parameters.AllowEmptyBeneficiaryAddress == true`, we can still set beneficiary address